src/lib.rs

```diff
@@ -470,6 +470,8 @@ pub mod pallet {
 					claimed_reward: initial_payment,
 				};
 
+				// It should be safe not to use saturating_add here
+				// as we already checked before that these rewards do not overflow existing ones
```

Comment:
You could consider using an `assert` here to express this. That may or may not be a good idea :)